datasets/MHCLG-Statutory-homelessness-Detailed-local-authority-level/main.py

```diff
@@ -2,14 +2,1854 @@
+    tableset = messytables.excel.XLSTableSet(fileobj = excel_obj)
+    tabs = list(xypath.loader.get_sheets(tableset, "*"))
+
+datasetTitle = original_tabs.title
+datasetTitle
+for tab in tabs:
+    print(tab.name)
+
+# + endofcell="--"
+for tab in tabs:
+    columns=['TO DO']
+    trace.start(datasetTitle, tab, columns, original_tabs.downloadURL)
+    if tab.name in ['A1']: #only transforming tab A1 for now
+    
+        remove_notes = tab.filter(contains_string('Notes')).expand(DOWN).expand(RIGHT)
+        ons_geo = tab.excel_ref('A6').expand(DOWN).is_not_blank() - remove_notes
+        period = tab.excel_ref('A1').is_not_blank() #period can be extracted from this cell 
+        sheet_name = tab.name
+        temp_assessment_duty_type_1 = tab.excel_ref('C3').expand(RIGHT)
+        temp_assessment_duty_type_2 = tab.excel_ref('C4').expand(RIGHT)
+        temp_assessment_duty_type_3 = tab.excel_ref('C5').expand(RIGHT)
+        observations = tab.excel_ref('C6').expand(DOWN).expand(RIGHT).is_not_blank() - remove_notes
+        
+        dimensions = [
+            HDim(ons_geo,'ONS Geography Code',DIRECTLY,LEFT),
+            HDim(period,'Period',CLOSEST,ABOVE),
+            HDim(temp_assessment_duty_type_1,'temp_assessment_duty_type_1',DIRECTLY, ABOVE),
+            HDim(temp_assessment_duty_type_2,'temp_assessment_duty_type_2',DIRECTLY, ABOVE),
+            HDim(temp_assessment_duty_type_3,'temp_assessment_duty_type_3',DIRECTLY, ABOVE),
+            #HDimConst("sheet_name", sheet_name) #Might be handy to have for post processing when other tabs are running also 
+        ]
+        tidy_sheet = ConversionSegment(tab, dimensions, observations)
+        savepreviewhtml(tidy_sheet, fname= tab.name + "PREVIEW.html")
+        trace.with_preview(tidy_sheet)
+        trace.store("combined_dataframe", tidy_sheet.topandas())
+    else:
+        continue 
+
+# # +
+#Below is how I matched the values up - I just took this info from the spec for tab A1
+
+#Initial Assessments                                                   Duty Type Owned
+#- Total Initial Assessments                                           - N/A
+#- Total owed a prevention of relief duty                              - Relief
+#- Threatened with homelessness within 56 days                         - Prevention
+#- Due to service of valid Section 21 Notice                           - Prevention
+#- Homeless                                                            - Relief
+#- Not homeless nor threatened with homelessness within 56 days        - No duty owed
+#- Number of Households in area (000s)                                 - All
+#- Households assessed as threatened with homelessness per(000s)       - N/A
+#- Households assessed as homeless per(000s)                           - N/A
+
+# # +
+df = trace.combine_and_trace(datasetTitle, "combined_dataframe")
+df
+df.rename(columns={'OBS' : 'Value', 'DATAMARKER' : 'Marker'}, inplace=True)
```

Comment:
Your data frame tailoring needs to be done at the end all at once if you're combining the data frame. YOu're doing a lot of work over and over and over and over and over again when you can just do it one time at the end if the columns align.